pkg/services/baremetal/host/host.go

```diff
@@ -679,10 +712,6 @@ func (s *Service) analyzeSSHErrorRegistering(sshErr error) (isSSHTimeoutError, i
 		}
 		reterr = fmt.Errorf("wrong ssh key: %w", sshErr)
 	case sshclient.IsConnectionRefusedError(sshErr):
-		if !rebootTriggered {
-			// Reboot did not trigger
-			return false, false, nil
-		}
```

Comment:
@janiskemper This is the only "real" change in the PR.  All other changes just improve error messages or create events.

AFAIK we have been both looking at that change. Does that look good to you?

Reply:
I'd like to spend some time thinking about possible issues. I have this on my list!

Reply:
I think we might have the following problem: 
In case we trigger a reboot but that reboot fails somehow (either API issue of Hetzner or ssh command didn't properly work), then we catch this here right now. This works because a connection refused error comes when the port is wrong. If the reboot didn't even start and we think it did, we try e.g. to go into the rescue mode with port 22, but we are currently not in rescue mode and the port is different. Then we get connection refused. This case is handled here, because if we realize that the reboot is not even triggered, then we can trigger it again.

If we just remove it, then we will not catch this case and I am not sure how important it is. We can either remove this now completely and then just see what happens, or we can add something to the current code. This addition would ensure that we have tried our best to trigger the reboot and it still didn't work, therefore we probably have a permament error. 

We can solve this by adding `if !rebootTriggered && s.scope.HetznerBareMetalHost.Spec.Status.ErrorType != infrav1.ErrorTypeHardwareRebootTriggered {`. 

If we do that, we make sure that we actually triggered the "strongest" form of reboot and it still doesn't work. 

At the same time, we will not ignore it forever and only until we actually triggered a hardware reboot. If this hardware reboot still didn't work, then we will go into the connection refused error and catch it properly

Reply:
@janiskemper thank you. Makes sense. I added your if-statement.